### PR TITLE
Update dependencies and enable new proposals.

### PR DESCRIPTION
**Pull request by milesj**
to: @lencioni 

- ~Updates our browser targets to the minimum requirements that Airbnb supports.~ (Follow up PR)
- Enables the optional `catch` and numeric separator proposals.
- Updates dependencies to their latest version, to ensure we're inheriting the latest code, and to dedupe easier.

### CODE REVIEW

**Comment by ljharb** [comment]
The targets change is a breaking change; the new syntax is a minor; any chance those could be split off, so the minor can be released before updating the targets?

File: package.json [hunk 1]
```diff
@@ -16,20 +16,22 @@
...
-    "@babel/preset-react": "^7.0.0",
+    "@babel/helper-plugin-utils": "^7.8.3",
+    "@babel/plugin-proposal-nullish-coalescing-operator": "^7.8.3",
+    "@babel/plugin-proposal-numeric-separator": "^7.8.3",
+    "@babel/plugin-proposal-object-rest-spread": "^7.9.0",
+    "@babel/plugin-proposal-optional-catch-binding": "^7.8.3",
+    "@babel/plugin-proposal-optional-chaining": "^7.9.0",
+    "@babel/plugin-transform-classes": "^7.9.2",
+    "@babel/plugin-transform-exponentiation-operator": "^7.8.3",
+    "@babel/plugin-transform-jscript": "^7.8.3",
```
Comment (esprehn) [inline package.json:27]: I think we should consider removing this one. It's trying to work around IE8 bugs, but it super bloats the code: https://kangax.github.io/nfe/#jscript-bugs

I doubt Chakra has any of those bugs (also I'm pretty sure we disable this internally)

Comment (milesj) [reply package.json:27]: Im cool with putting this + browser targets in another PR so we can group a breaking change.

**Comment by milesj** [comment]
@ljharb Done. Will do browsers in another PR.

**Review by ljharb — APPROVED** [review]
LGTM as a semver-minor

**Pull request merged**